arviz/data/inference_data.py

```diff
@@ -125,14 +127,38 @@ def __init__(self, **kwargs):
                     self._groups_warmup.append(key)
 
     def __repr__(self):
-        """Make string representation of object."""
+        """Make string representation of InferenceData object."""
         msg = "Inference data with groups:\n\t> {options}".format(
             options="\n\t> ".join(self._groups)
         )
         if self._groups_warmup:
             msg += "\n\nWarmup iterations saved ({}*).".format(WARMUP_TAG)
         return msg
 
+    def _repr_html_(self):
+        """Make html representation of InferenceData object."""
+        display_style = OPTIONS["display_style"]
```

Comment:
> > I am going to request tests before merge. IMO all code should be tested if it can, and this is no exception :). I would add a test that calls `_repr_html_` directly and tests that the html string is returned.
> > I would also add a tests the correct repr is returned without an explicit call. I assume this will require some mocking and to "trick" the non jupyter notebook test to provide the html repr.
> 
> I have added the basic test. How do I add the second test? I looked through but couldn't find the way.

This is going to be  fun one. Making tests is sometimes a puzzle and harder than the actual programming itself, but important if you want you feature to be stable and not broken by any others.

In the test you'll have to monkeypatch `OPTIONS` to so when the test executes the value is `text` to test that this if condition works. Just being up front here it's going to be hard and confusing, but youre going to learn a lot more about python.

I don't want to block this whole PR for that one thing though. We can merge but I really would like you to add this test in a followup PR. At a minimum create an issue so we can track


https://docs.pytest.org/en/latest/monkeypatch.html

Reply:
I think it will not be necessary to monekypatch because the display style option is already customizable similarly to an rcParam. We can start the test by setting the `display_style` to `text`/`html`, make sure the correct repr is returned and then finish the test by going back to the original displaystyle (much like your original implementation of the repr @percygautam).

Reply:
Its typically bad practice to have tests rely on global state like rcparam although in this case lets just do it cause monkeypatching would be annoying :D

Reply:
I am okay with both options. We can currently implement the test using rcParams, and create the issue to track the monkeypatch test and merge this PR. I will implement the test later.
The `html_repr` really facilitates to check other methods that I am adding, so it'll be better to merge this one as soon as possible.

Reply:
Agree with merging as soon as possible! Seems like youre only a couple linting errors away :)